src/Host.cpp

```diff
@@ -1147,6 +1147,38 @@ bool Host::installPackage( const QString& fileName, int module )
     }
     // reorder permanent and temporary triggers: perm first, temp second
     mTriggerUnit.reorderTriggersAfterPackageImport();
+
+    // This event was inspired by "Zoilder" in a Mudlet derivative in 2015
+    // Their version appended the packageName and "Event" to
+    // "sysInstallPackage", but I have revised it to pass the packageName as a
+    // separate argument.
+    // Due to the range of circumstances in which this method can be called
+    // I have also included:
+    // * the invoking "module" argument in case it is necesary to differentiate
+    // the way in which the installPackage code is called - I think this could
+    // be a factor in some use cases that Zoilder would not have experienced in
+    // their usage.
+    // * the original package filename
+    //
+    // The originators comment was (ES):
+    // Se ejecuta el evento que indica que se ha instalado el paquete.
+    // Así el propio paquete puede auto-inicializarse sin tener que reabrir la
+    // sesión.
+    // which I think translates to (EN):
+    // The event that indicates that the package is installed runs.
+    // Thus the package itself can auto-initialize without having to reopen the
+    // session.
+    TEvent installEvent;
+    installEvent.mArgumentList.append( QStringLiteral( "sysInstallPackageEvent" ) );
+    installEvent.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
+    installEvent.mArgumentList.append( packageName );
+    installEvent.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
+    installEvent.mArgumentList.append( QString::number( module ) );
```

Comment:
:wave: @Chris7  - good to see you!

Checking the source I see that the use of those values 0 to 3 is as stated - and I think package/module writers _may_ want to be aware of the differences between them as it _could_ influence the behaviour they want to take if the package/module being added/removed is something they are concerned with.  For instance one module could be installing another itself - so would expect to see a "3" type install happening and would not expect to have to handle it specially; but if it saw the same module with a "1" it would know that either the system is being started up or the user has just added the interesting module themselves...

As for "naming" the type of the event, to be honest from what scripting I have done number codes are quite usable and (for a small range of numbers) they ARE easier to translate at least in those parts of the world using "Western Arabic" numerals :grinning: